Mathlib/Topology/Algebra/GroupCompletion.lean

```diff
@@ -115,101 +126,129 @@ instance : AddMonoid (Completion α) :=
+    mul_left_inv := fun a ↦
       Completion.induction_on a
         (isClosed_eq (continuous_map₂ Completion.continuous_map continuous_id) continuous_const)
         fun a ↦
-        show -(a : Completion α) + a = 0 by
-          rw_mod_cast [add_left_neg]
+        show (a : Completion α)⁻¹ * a = 1 by
+          rw_mod_cast [mul_left_inv]
           rfl }
 
-instance uniformAddGroup : UniformAddGroup (Completion α) :=
-  ⟨uniformContinuous_map₂ Sub.sub⟩
+@[to_additive]
+noncomputable instance uniformGroup : UniformGroup (Completion α) :=
+  ⟨uniformContinuous_map₂ Div.div⟩
 
-instance {M} [Monoid M] [DistribMulAction M α] [UniformContinuousConstSMul M α] :
-    DistribMulAction M (Completion α) :=
-  { (inferInstance : MulAction M <| Completion α) with
-    smul_add := fun r x y ↦
-      induction_on₂ x y
-        (isClosed_eq ((continuous_fst.add continuous_snd).const_smul _)
-          ((continuous_fst.const_smul _).add (continuous_snd.const_smul _)))
-        fun a b ↦ by simp only [← coe_add, ← coe_smul, smul_add]
-    smul_zero := fun r ↦ by rw [← coe_zero, ← coe_smul, smul_zero r] }
-
-/-- The map from a group to its completion as a group hom. -/
-@[simps]
-def toCompl : α →+ Completion α where
+/-- The map from a multiplicative group to its completion as a group hom. -/
+@[to_additive (attr := simps) "The map from an additive group to its completion as a group hom."]
+def toComplMulHom : α →* Completion α where
   toFun := (↑)
-  map_add' := coe_add
-  map_zero' := coe_zero
-#align uniform_space.completion.to_compl UniformSpace.Completion.toCompl
+  map_mul' := coe_mul'
+  map_one' := coe_one'
+#align uniform_space.completion.to_compl UniformSpace.Completion.toComplAddHom
+
+@[deprecated] alias toCompl := toComplAddHom
 
-theorem continuous_toCompl : Continuous (toCompl : α → Completion α) :=
+@[to_additive]
+theorem continuous_toComplMulHom : Continuous (toComplMulHom : α →* Completion α) :=
   continuous_coe α
-#align uniform_space.completion.continuous_to_compl UniformSpace.Completion.continuous_toCompl
+#align uniform_space.completion.continuous_to_compl UniformSpace.Completion.continuous_toComplMulHom
+
+@[deprecated] alias continuous_toCompl := continuous_toComplAddHom
 
 variable (α)
 
-theorem denseInducing_toCompl : DenseInducing (toCompl : α → Completion α) :=
+@[to_additive]
+theorem denseInducing_toComplMulHom : DenseInducing (toComplMulHom : α →* Completion α) :=
   denseInducing_coe
-#align uniform_space.completion.dense_inducing_to_compl UniformSpace.Completion.denseInducing_toCompl
+#align uniform_space.completion.dense_inducing_to_compl UniformSpace.Completion.denseInducing_toComplMulHom
+
+@[deprecated] alias denseInducing_toCompl := denseInducing_toComplAddHom
 
 variable {α}
 
+end UniformGroup
+
+section UniformAddGroup
+
+variable [UniformSpace α] [AddGroup α] [UniformAddGroup α]
+
+instance {M} [Monoid M] [DistribMulAction M α] [UniformContinuousConstSMul M α] :
+    DistribMulAction M (Completion α) :=
```

Comment:
You'll have to write it by hand, but this multiplicativizes as `MulDistribMulAction`.